src/ui/public/agg_types/buckets/histogram.js

```diff
@@ -3,13 +3,62 @@ import moment from 'moment';
 import 'ui/validate_date_interval';
 import AggTypesBucketsBucketAggTypeProvider from 'ui/agg_types/buckets/_bucket_agg_type';
 import AggTypesBucketsCreateFilterHistogramProvider from 'ui/agg_types/buckets/create_filter/histogram';
+import FilterBarQueryFilterProvider from 'ui/filter_bar/query_filter';
 import intervalTemplate from 'ui/agg_types/controls/interval.html';
 import minDocCountTemplate from 'ui/agg_types/controls/min_doc_count.html';
 import extendedBoundsTemplate from 'ui/agg_types/controls/extended_bounds.html';
+import autoScaleTemplate from 'ui/agg_types/controls/auto_scale.html';
 export default function HistogramAggDefinition(Private) {
   let BucketAggType = Private(AggTypesBucketsBucketAggTypeProvider);
   let createFilter = Private(AggTypesBucketsCreateFilterHistogramProvider);
+  let queryFilter = Private(FilterBarQueryFilterProvider);
+  const NUM_BUCKETS = 50;
 
+  function numberOfDigits(num) {
+    return (Math.trunc(num) + '').length;
+  }
+
+  function roundToNearest(num, roundDigit) {
```

Comment:
If I understand the purpose of this function correctly, I think you could just use Lodash's `round` with a negative precision https://lodash.com/docs/3.10.1#round